### PR TITLE
Update scrypto dep

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
To `develop-baf0142b2`.

### CODE REVIEW

**Comment by jakrawcz-rdx** [comment]
⚠️ WIP ⚠️ 

This PR is WIP builds, but most likely maaaany tests fail, since I left a few TODOs for cases where I was not sure how to proceed.

These TODOs are marked with `// TODO(code review): <details>`. They are related to:
- static `ACCOUNT_PACKAGE_ADDRESS` no longer existing, but still used (at least in tests?) - @talekhinezh can you advice?
- new `Account` substate type: I assumed it is supposed to be propagated up to the core API, but I have no idea about its representation - @dhedey?
- introduction of `NodeModuleId`:
  - in many places I just inserted `::SELF` - @talekhinezh please verify
  - also, maybe this one has to somehow reach the core API? - @dhedey please see my question regarding `MappedSubstateId`
- the `new_account_with_resource` got deleted, and I assume that now a `new_account` + `deposit entire worktop` combo is supposed to be used, but I am too lame to write a working piece of manifest like that - @talekhinezh I will appreciate your help!

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/addressing.rs:229]: My preference here is yes so that there is little abstraction between the Core API and the underlying ledger state but I think it mostly depends on what is the Gateway consumer ready to handle.

File: core-rust-bridge/src/main/java/com/radixdlt/rev2/ScryptoConstants.java
Comment (talekhinezh) [inline core-rust-bridge/src/main/java/com/radixdlt/rev2/ScryptoConstants.java:83]: There is no address currently but there will be one in a soon to come PR on the scrypto side. For now, this can be removed.

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/addressing.rs:259]: This looks fine.

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/addressing.rs:288]: This is fine.

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/addressing.rs:326]: This is fine.

File: core-rust/core-api-server/src/core_api/conversions/receipt.rs
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/conversions/receipt.rs:191]: Sounds okay to me.

File: core-rust/core-api-server/src/core_api/generated/models/substate.rs
Comment (talekhinezh) [inline core-rust/core-api-server/src/core_api/generated/models/substate.rs:18]: This should be generated by updating the `core-api-schema.yaml` file and then running:
```
python3 core-rust/core-api-server/scripts/generate-openapi-server.py
```

I would suggest following the same pattern used by AccessController.

File: core-rust/state-manager/src/transaction/builder.rs
Comment (talekhinezh) [inline core-rust/state-manager/src/transaction/builder.rs:110]: ooo this may be impossible with the current setup since there is no way to get a reference to the new account just created.

I'd either suggest holding off on the PR until the scrypto side gets updated, or ignoring the tests which currently use this functionality, or update the tests which use to use virtual accounts instead of new accounts.

File: core-rust/core-api-server/src/core_api/generated/models/substate.rs
Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/generated/models/substate.rs:18]: > This should be generated

Huh, I guess the `/generated/` in the path is not enough of a hint for me 🤷‍♂️ 

> following the same pattern used by AccessController

I interpret this as "literally return the same `data_struct`" (which reveals I have no clue about that api) 👍

File: core-rust/core-api-server/src/core_api/helpers.rs
Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/helpers.rs]: ah, thanks, I now tried to pass my best guesses, PTAL.

File: core-rust/core-api-server/src/core_api/conversions/addressing.rs
Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/conversions/addressing.rs]: alright, I consulted @KrzysztofLabus-RDX and we came up with `module_type` name for the new field; please see if this mapping is good enough.

File: core-rust/state-manager/src/transaction/builder.rs
Comment (jakrawcz-rdx) [reply core-rust/state-manager/src/transaction/builder.rs]: AFAIS there is only 1 test relying on this, so I guess it doesn't hurt to ignore it temporarily 👍

File: core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java
Comment (dhedey) [inline core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java:136]: I'd suggest creating a `createNewVirtualAccountWithXrdTransaction(publicKey)` via using:
```
var virtualAccountAddress = Address.virtualAccountAddress(publicKey);
...
```

And a manifest which transfers XRD to the virtual account, thereby creating it:
```
CALL_METHOD
    ComponentAddress("${virtualAccountAddress}") 
    "deposit_batch"
    Expression("ENTIRE_WORKTOP");
```

**Review by dhedey — APPROVED** [review]
Really nice work. Happy to merge once the package address issue is resolved.

File: core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java
Comment (jakrawcz-rdx) [reply core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java:136]: The "virtual account" snippets are super helpful for adjusting many of the failing tests, thanks 👍 

However this one here is named `new_account_creates_transfer_of_xrd_to_account`, which suggests that it wants to test exactly the "new account with resource" function, which is no longer available (in fact, this is the only test that calls Rust jni `create_new_account_intent_bytes()` method).

If I adjust the `create_new_account_intent_bytes()` to use virtual account, then it would mean renaming it (to e.g. `deposit_xrd_to_virtual_account_intent_bytes()`) and also renaming the test to `can_transfer_xrd_to_account`. I could do it, but it is something already tested by other tests.

So I think we are left with options:
A) Keep the test `@Ignore`d temporarily, since Josh suggests that "scrypto side will get updated" (making this function available [and worth testing] again, I assume).
B) Remove the test and the entire jni `create_new_account_intent_bytes()`.

**Review by KrzysztofLabus-RDX — APPROVED** [review]

File: core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java
Comment (dhedey) [reply core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java:136]: I'm not sure (A) will be added back, and if it is, I don't think it will be tested here, so happy with (B).

That said, I think this might be the only test of the `get_component_resources` function - but that's just a function for use by tests anyway so I don't think it's a problem.

Comment (jakrawcz-rdx) [reply core/src/test/java/com/radixdlt/rev2/REv2ConsensusTransferTest.java:136]: if I am looking at the right `get_component_resources`, then it is still used by other tests (via `get_component_xrd`).

So with no remorse, I deleted this test 🤷‍♂️

**Comment by jakrawcz-rdx** [comment]
I've been able to fix most of the tests. There are 2 mysteries left, both marked with `TODO(code review)` right now in this PR:

1. `state_reader_on_genesis_returns_correct_amounts()` - we expect X staked, but 0 gets staked. @LukasGasior1 might be able to help?

2. `test_core_api_can_submit_and_commit_transaction()` - suddenly `streamTransactionsPost()` started throwing 500 (this is the only test trying to do that, AFAIS). Looks treatable, I just gotta learn how to enter with some debug / extensive logging there.

**Comment by jakrawcz-rdx** [comment]
The mystery 1. of https://github.com/radixdlt/babylon-node/pull/304#issuecomment-1424271184 is clarified a bit: our test util for reading out account balances does not know about the new `Account` substate, and it now reports 0 (thanks @talekhinezh for confirming!). I am on the right track to fix it 👍

**Comment by jakrawcz-rdx** [comment]
Finally, all tests pass 🎉

**Pull request merged**
